pkg/pub_worker/lib/src/bin/dartdoc_wrapper.dart

```diff
@@ -85,14 +85,39 @@ Future<void> postProcessDartdoc({
     _log.info('Finished .tar.gz archive');
   });
 
-  await Future.wait([
-    processingFuture,
-    archiveFuture,
-  ]);
+  try {
+    await Future.wait([
+      processingFuture,
+      archiveFuture,
+    ]);
 
-  // Move from temporary output directory to final one, ensuring that
-  // documentation files won't be present unless all files have been processed.
-  // This helps if there is a timeout along the way.
-  await Directory(tmpOutDir).rename(p.join(outputFolder, 'doc'));
-  await tmpTar.rename(p.join(outputFolder, 'doc', 'package.tar.gz'));
+    // Move from temporary output directory to final one, ensuring that
+    // documentation files won't be present unless all files have been processed.
+    // This helps if there is a timeout along the way.
+    await tmpOutDir.rename(p.join(outputFolder, 'doc'));
+    await tmpTar.rename(p.join(outputFolder, 'doc', 'package.tar.gz'));
+  } catch (e, st) {
+    // We are likely out of free disk space.
+    //
+    // One of the isolate has already stopped, the other may be still working.
```

Comment:
That's not the semantics of [`Future.wait`](https://api.dart.dev/stable/3.5.3/dart-async/Future/wait.html).

Reply:
Good catch. Fixed it. We may wait for both, as it is likely that both will be running out of the free space the same time.